src/components/pages/admin/contacts/RoleEditorItem.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <div class="mt-3 rounded border border-primary-20 bg-primary-10">
+    <div class="flex items-center bg-primary-5 px-4 py-1 text-sm">
+      <AppRoundBadge :type="isRoleCurrent(role) ? 'success' : 'danger'" />
+      <strong class="mx-2 font-bold uppercase text-body-80">
+        {{ t(`common.role.${role.role}`) }}
+      </strong>
+      <span>{{ formatLocale(role.dateAdded, 'P') + ' → ' }}</span>
+      {{
+        role.dateExpires
+          ? formatLocale(role.dateExpires, 'P')
+          : t('contacts.data.rolesCopy.today')
+      }}
+      <div class="ml-auto">
+        <AppButton variant="text" size="sm" @click="formVisible = true"
+          >Edit</AppButton
+        >
+        <AppButton variant="dangerText" size="sm" @click="handleDeleteRole">
+          Delete
+        </AppButton>
+      </div>
+    </div>
+
+    <div v-if="formVisible" class="flex flex-1 p-4">
+      <form @submit.prevent="handleFormSubmit" class="flex-initial">
```

Comment:
I guess the form could probably be extracted to it's own component so it can be shared between add/edit